generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx

```diff
@@ -21,30 +21,59 @@ import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { Button } from 'reactstrap';
 import { Translate, ICrudGetAction, TextFormat } from 'react-jhipster';
-import { FaPlus, FaEye, FaPencil, FaTrash } from 'react-icons/lib/fa';
+import { FaPlus, FaEye, FaPencil, FaSort, FaTrash } from 'react-icons/lib/fa';
 
 import { getUsers } from '../../../reducers/user-management';
 import { APP_DATE_FORMAT } from '../../../config/constants';
+import { ITEMS_PER_PAGE } from '../../../shared/util/pagination.constants';
+import { getItemsNumber, getSortState, IPaginationState } from '../../../shared/util/pagination-utils';
+import { PaginationComponent } from '../../../shared/util/pagination-component';
 
 export interface IUserManagementProps {
   getUsers: ICrudGetAction;
   users: any[];
   account: any;
   match: any;
+  totalItems: 0;
+  history: any;
+  location: any;
 }
 
-export class UserManagement extends React.Component<IUserManagementProps, undefined> {
+export class UserManagement extends React.Component<IUserManagementProps, IPaginationState> {
 
   constructor(props) {
     super(props);
+    this.state = {
+      ...getSortState(props.location)
+    };
   }
 
   componentDidMount() {
-    this.props.getUsers();
+    this.props.getUsers(this.state.activePage - 1, this.state.itemsPerPage, `${this.state.sort},${this.state.order}`);
```

Comment:
can we extract this line to a method as below and call the method here and in other places?
```
getUsers = () => {
  const { activePage, itemsPerPage, sort, order } = this.state;
  this.props.getUsers(activePage - 1, itemsPerPage, `${sort},${order}`);
}
```